File: templates/web-python-playwright/pages/login_page.py
"""Страница логина."""
import allure
from playwright.sync_api import Locator

from pages.base_page import BasePage


class LoginPage(BasePage):
    path = "/login"

    @property
    def email_input(self) -> Locator:
        return self.page.locator("#login-email-input")

    @property
    def password_input(self) -> Locator:
        return self.page.locator("#login-password-input")

    @property
    def submit_button(self) -> Locator:
        return self.page.locator("button[type=submit]")

    @property
    def error_message(self) -> Locator:
        return self.page.locator("#login-error")

    def login(self, email: str, password: str) -> None:
        with allure.step(f"Логинимся через форму: {email}"):
            self.email_input.fill(email)
            self.password_input.fill(password)
            self.submit_button.click()

File: templates/web-java-selenide/src/test/java/web/pages/LoginPage.java
package web.pages;

import com.codeborne.selenide.SelenideElement;
import io.qameta.allure.Step;

import static com.codeborne.selenide.Selenide.$;
import static com.codeborne.selenide.Selenide.open;

/** Страница логина. */
public class LoginPage {

    public final SelenideElement emailInput = $("#login-email-input");
    public final SelenideElement passwordInput = $("#login-password-input");
    public final SelenideElement submitButton = $("button[type=submit]");
    public final SelenideElement errorMessage = $("#login-error");

    @Step("Открываем страницу логина")
    public LoginPage openPage() {
        open("/login");
        return this;
    }

    @Step("Логинимся через форму: {email}")
    public void login(String email, String password) {
        emailInput.setValue(email);
        passwordInput.setValue(password);
        submitButton.click();
    }
}
